src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -580,6 +580,12 @@ def sigalarm_handler(signum, frame):
                            {"$set": {"lm_info"  : lrms.lm_info.get('version_info'),
                                      "lm_detail": lrms.lm_info.get('lm_detail')}})
 
+            # OSG Specific
+            osg_resource_name = os.environ.get('GLIDEIN_ResourceName')
+            if osg_resource_name:
+                mongo_p.update({"_id": pilot_id},
+                               {"$set": {"osg_resource_name": osg_resource_name}})
```

Comment:
See https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/compute_pilot.py#L290 - is basically exposes backend specific pilot information for user consumption.  The Spark LM for example exposes the spark service URL in `pilot.resource_detail['lm_detail'].

But no  worries - since you put the information into the DB, the exposure via API etc. is orthogonal.